consul-lambda-registrator/proxy/proxy.go

```diff
@@ -47,6 +48,7 @@ func (s *Server) Serve() error {
 	connErrChan := make(chan error)
 	lwg := &sync.WaitGroup{}
 
+	s.lmu.Lock()
```

Comment:
I think there is a chance this mutex never gets unlocked if starting a listener fails on line 61. Does it make sense to put the unlock in a defer?

Reply:
We can't `defer` the unlock here because `Serve` blocks (until there is an error or the proxy is closed). So a defer will make it so we don't release the lock until the proxy is closed. All of the func's in the `for` loop are started in separate go-routines so the return statement on line 61 just returns from that goroutine, not the `Serve` method. Basically it is:

```go
s.lmu.Lock()
for each config {
   // start go routines
}
s.lmu.Unlock()
```